pro_tes/middleware/middleware_pipeline.py

```diff
@@ -0,0 +1,128 @@
+"""Set up the middleware pipeline."""
+import importlib
+import logging
+
+from flask import current_app
+
+logger = logging.getLogger(__name__)
+
+
+class MiddlewarePipeline:
+    """Middleware pipeline."""
+
+    def __init__(self, middlewares=None):
+        """Construct object instance."""
```

Comment:
Please add type hints at least for method arguments and return values, ideally also for all other variables